AppCenter/AppCenter/MSAppCenter.m

```diff
@@ -174,6 +178,57 @@ + (void)setCustomProperties:(MSCustomProperties *)customProperties {
 }
 #endif
 
++ (void)setAuthToken:(NSString *)authToken {
+  [[MSAppCenter sharedInstance] setAuthToken:authToken];
+}
+
+- (void)setAuthToken:(NSString *)authToken {
+  MSAuthTokenContext *authTokenContext = [MSAuthTokenContext sharedInstance];
+  MSJwtClaims *claims = nil;
+  if (authToken != nil) {
+    claims = [MSJwtClaims parse:authToken];
+  }
+  if (claims != nil) {
+    [authTokenContext setAuthToken:authToken withAccountId:claims.subject expiresOn:claims.expiration];
+    MSLogDebug(MSAppCenter.logTag, @"Authentication token has been refreshed.");
+  } else {
+    [authTokenContext setAuthToken:nil withAccountId:nil expiresOn:nil];
+    MSLogDebug(MSAppCenter.logTag, @"Removed authentication token.");
+  }
+}
+
++ (void)setAuthTokenDelegate:(id<MSAuthTokenDelegate>)authTokenDelegate {
+  [[MSAppCenter sharedInstance] setAuthTokenDelegate:authTokenDelegate];
+}
+
+- (void)setAuthTokenDelegate:(id<MSAuthTokenDelegate>)authTokenDelegate {
+  MSAuthTokenContext *authTokenContext = [MSAuthTokenContext sharedInstance];
+  @synchronized(self.authTokenContextDelegateWrapper) {
+    if (authTokenDelegate != nil) {
+      self.authTokenContextDelegateWrapper = [[MSAuthTokenContextDelegateWrapper alloc]
+           initWithAuthTokenDelegate:authTokenDelegate
+          authTokenCompletionHandler:^(NSString *jwt) {
+            MSJwtClaims *claims = [MSJwtClaims parse:jwt];
```

Comment:
The completion block should use the instance auth set token method, supposed to be same logic. At least on Android that's what we did.